packages/liveblocks-react-comments/src/utils/url.ts

```diff
@@ -3,8 +3,8 @@ const ABSOLUTE_URL_REGEX = /^[a-zA-Z][a-zA-Z\d+\-.]*?:/;
 
 export function generateURL(
   url: string,
-  params: Record<string, string | number | undefined>,
-  hash: string
+  params: Record<string, string | number | undefined> = {},
```

Comment:
I was thinking more to keep it `undefined` as a regular optional argument, and to skip the `for...of` loop when it is `undefined`.

Reply:
I thought they were effectively the same thing. That being said, I do agree the code would be more readable if I kept it undefined and performed any kind of undefined check inside the function. Updated to the following:

```ts
export function generateURL(
  url: string,
  params?: Record<string, string | number | undefined>,
  hash?: string
) {
  const isAbsolute = ABSOLUTE_URL_REGEX.test(url);
  const urlObject = new URL(url, isAbsolute ? undefined : PLACEHOLDER_BASE_URL);

  if (params !== undefined) {
    for (const [param, value] of Object.entries(params)) {
      if (value) {
        urlObject.searchParams.set(param, String(value));
      }
    }
  }

  // Only add the new hash if the URL does not already have one
  if (!urlObject.hash && hash !== undefined) {
    urlObject.hash = `#${hash}`;
  }

  return isAbsolute
    ? urlObject.href
    : urlObject.href.replace(PLACEHOLDER_BASE_URL, "");
}
```